src/Containers/Microsoft.NET.Build.Containers/ContainerHelpers.cs

```diff
@@ -143,13 +144,27 @@ internal static bool IsValidImageTag(string imageTag)
         return ReferenceParser.anchoredTagRegexp.IsMatch(imageTag);
     }
 
+
     /// <summary>
     /// Given an already-validated registry domain, this is our hueristic to determine what HTTP protocol should be used to interact with it.
+    /// If the domain is localhost, we default to HTTP. Otherwise, we check the Docker config to see if the registry is marked as insecure.
     /// This is primarily for testing - in the real world almost all usage should be through HTTPS!
     /// </summary>
     internal static Uri TryExpandRegistryToUri(string alreadyValidatedDomain)
     {
-        var prefix = alreadyValidatedDomain.StartsWith("localhost", StringComparison.Ordinal) ? "http" : "https";
+        string prefix = "https";
+        if (alreadyValidatedDomain.StartsWith("localhost", StringComparison.Ordinal))
+        {
+            prefix = "http";
+        }
+#if !NET472
+        //check the docker config to see if the registry is marked as insecure
+        else if (DockerCli.IsInsecureRegistry(alreadyValidatedDomain))
+        {
+            prefix = "http";
+        }
+#endif
```

Comment:
Why is this conditional necessary? Is the DockerCli not available on the .NET Framework build?

Reply:
When I first start this PR I found it build failed on net472 but passed on net8 when I add the code, so I add the conditional.

I am not sure whether it is nessary now, I will check later.

Reply:
@baronfel That's true. DockerCli is not available on the .NET framework build.

Microsoft.NET.Build.Containers.csproj
```xml
  <!-- net472 builds manually import files to compile -->
  <ItemGroup Condition="'$(TargetFramework)' == 'net472'">
    <Compile Remove="**/*.*" />
    <Compile Include="ReferenceParser.cs" />
    <Compile Include="KnownStrings.cs" />
    <Compile Include="Tasks/ParseContainerProperties.cs" />
    <Compile Include="Tasks/CreateNewImage.Interface.cs" />
    <Compile Include="Tasks/CreateNewImageToolTask.cs" />
    <Compile Include="Tasks/ComputeDotnetBaseImageAndTag.cs" />
    <Compile Include="ContainerHelpers.cs" />
    <Compile Include="net472Definitions.cs" />
    <Compile Include="VSHostObject.cs" />
    <Compile Include="Port.cs" />
    <Compile Include="Resources\Resource.cs" />
    <Compile Include="Resources\Strings.Designer.cs" />
    <Compile Include="Globals.cs" />
  </ItemGroup>
```

When I run build after remove the condition, net472 build failed.
<img width="1193" alt="image" src="https://github.com/dotnet/sdk/assets/4465571/b117ff0a-27c2-434b-aa3d-655ccee1f0eb">

Reply:
Ok, I've pushed a small commit that adds DockerCli.cs to the net472 build while also conditionally-compiling the parts that are not relevant for this new feature specifically for .NET. I removed the ifdef for the ContainerHelpers and it's all compiling!